### PR TITLE
feat:Add very basic LMFS support to the debugger

### LINKED ISSUE #6: Add LMFS Support
Fleetdebugger currently only supports ODRD vehicles & trips.   Need to ensure that delivery vehicle and task support actually works.

### PR DESCRIPTION
**Pull request by greghutch**
Changes:
   - Refactored slightly how logs are gathered.
   - Doesn't forec use to indicate whether vehicle being
     debugged is a delivery or ridesharing vehicle
   - very basic plumbing on the visualization to get the route
     path
   - haven't yet updated the log viewer or other components

Fixes #6

### CODE REVIEW

File: dune-buggy.js [hunk 1]
```diff
@@ -64,49 +68,122 @@ async function fetchTripLogsForVehicle(vehicleLogs, vehicle_id) {
...
-    console.log(`no trips associated with vehicle ${argv.vehicle}`);
+    console.log(`no trips associated with vehicle id ${argv.vehicle}`);
   }
   return tripLogs;
 }
 
-let label, labelVal;
+/*
+ * Extract task_ids from vehicle logs, and query again to
+ * get the createTrip/update trip calls that may not be labeled
```
Comment (ewbao) [inline dune-buggy.js:86]: Do you mean task?

File: dune-buggy.js [hunk 2]
```diff
@@ -64,49 +68,122 @@ async function fetchTripLogsForVehicle(vehicleLogs, vehicle_id) {
...
+  );
+  const taskLogs = await fetchTaskLogsForVehicle(
+    argv.vehicle,
+    deliveryVehicleLogs
+  );
+  const tripLogs = await fetchTripLogsForVehicle(argv.vehicle, vehicleLogs);
+
+  const solutionType = vehicleLogs.length === 0 ? "LMFS" : "ODRD";
+
+  const rawLogs = _(vehicleLogs)
```
Comment (ewbao) [inline dune-buggy.js:175]: Can we assume here that deliveryVehicle and vehicle ids are unique?

File: dune-buggy.js [hunk 3]
```diff
@@ -64,49 +68,122 @@ async function fetchTripLogsForVehicle(vehicleLogs, vehicle_id) {
...
+/*
+ * Extract task_ids from vehicle logs, and query again to
+ * get the createTrip/update trip calls that may not be labeled
+ * with a vehicle
+ */
+async function fetchTaskLogsForVehicle(vehicle_id, vehicleLogs) {
+  if (!vehicle_id) {
+    return [];
+  }
+  console.log("Loading tasks logs for delivery vehicle id", vehicle_id);
```
[the remark this answers is not in the document]
Comment (greghutch) [reply dune-buggy.js:93]: done

File: dune-buggy.js [hunk 4]
```diff
@@ -64,49 +68,122 @@ async function fetchTripLogsForVehicle(vehicleLogs, vehicle_id) {
...
+  console.log(`Fetching logs for ${label} = ${labelVal}`);
+  return await logging.fetchLogs(label, [labelVal], argv.daysAgo);
+}
+
+async function fetchDeliveryVehicleLogs(deliveryVehicle, vehicleLogs) {
+  if (vehicleLogs.length !== 0) {
+    // regular vehicle logs found, not a deliveryVehicle
+    return [];
+  }
+  // TODO: is it more efficient to run the log query twice
```
[the remark this answers is not in the document]
Comment (greghutch) [reply dune-buggy.js:129]: You're probably right.  I'll see how how long it takes before I need a separate command (or maybe even a separate debugger).

File: dune-buggy.js (hunk 1 above)
Comment (greghutch) [reply dune-buggy.js:86]: yes.  fixed.

File: dune-buggy.js (hunk 2 above)
Comment (greghutch) [reply dune-buggy.js:175]: Probably not.  I filed an issue for future cleanup: https://github.com/googlemaps/fleet-debugger/issues/66

**Pull request merged**